src/mpsc/errors.rs

```diff
@@ -88,6 +108,72 @@ impl<T> fmt::Display for Closed<T> {
 #[cfg(feature = "std")]
 impl<T> std::error::Error for Closed<T> {}
 
+// === impl SendTimeoutError ===
+
+impl SendTimeoutError {
+    #[cfg(feature = "std")]
+    pub(crate) fn with_value<T>(self, value: T) -> SendTimeoutError<T> {
+        match self {
+            Self::Timeout(()) => SendTimeoutError::Timeout(value),
+            Self::Closed(()) => SendTimeoutError::Closed(value),
+        }
+    }
+}
+
+impl<T> SendTimeoutError<T> {
+    /// Returns `true` if this error was returned because the channel is still
+    /// full after the timeout has elapsed.
+    pub fn is_timeout(&self) -> bool {
+        matches!(self, Self::Timeout(_))
+    }
+
+    /// Returns `true` if this error was returned because the channel has closed
+    /// (e.g. the [`Receiver`] end has been dropped).
+    ///
+    /// If this returns `true`, no future [`try_send`] or [`send`] operation on
+    /// this channel will succeed.
+    ///
+    /// [`Receiver`]: super::Receiver
+    /// [`try_send`]: super::Sender::try_send
+    /// [`send`]: super::Sender::send
+    /// [`Receiver`]: super::Receiver
+    pub fn is_closed(&self) -> bool {
+        matches!(self, Self::Timeout(_))
+    }
+
+    /// Unwraps the inner `T` value held by this error.
+    ///
+    /// This method allows recovering the original message when sending to a
+    /// channel has failed.
+    pub fn into_inner(self) -> T {
+        match self {
+            Self::Timeout(val) => val,
+            Self::Closed(val) => val,
+        }
+    }
+}
+
+impl<T> fmt::Debug for SendTimeoutError<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(match self {
+            Self::Timeout(_) => "SendTimeoutError::Timeout(..)",
+            Self::Closed(_) => "SendTimeoutError::Closed(..)",
+        })
+    }
+}
+
+impl<T> fmt::Display for SendTimeoutError<T> {
```

Comment:
```suggestion
#[cfg(feature = "std")]
impl SendTimeoutError {
    pub(crate) fn with_value<T>(self, value: T) -> SendTimeoutError<T> {
        match self {
            Self::Timeout(()) => SendTimeoutError::Timeout(value),
            Self::Closed(()) => SendTimeoutError::Closed(value),
        }
    }
}

#[cfg(feature = "std")]
impl<T> SendTimeoutError<T> {
    /// Returns `true` if this error was returned because the channel is still
    /// full after the timeout has elapsed.
    pub fn is_timeout(&self) -> bool {
        matches!(self, Self::Timeout(_))
    }

    /// Returns `true` if this error was returned because the channel has closed
    /// (e.g. the [`Receiver`] end has been dropped).
    ///
    /// If this returns `true`, no future [`try_send`] or [`send`] operation on
    /// this channel will succeed.
    ///
    /// [`Receiver`]: super::Receiver
    /// [`try_send`]: super::Sender::try_send
    /// [`send`]: super::Sender::send
    /// [`Receiver`]: super::Receiver
    pub fn is_closed(&self) -> bool {
        matches!(self, Self::Timeout(_))
    }

    /// Unwraps the inner `T` value held by this error.
    ///
    /// This method allows recovering the original message when sending to a
    /// channel has failed.
    pub fn into_inner(self) -> T {
        match self {
            Self::Timeout(val) => val,
            Self::Closed(val) => val,
        }
    }
}

#[cfg(feature = "std")]
impl<T> fmt::Debug for SendTimeoutError<T> {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        f.write_str(match self {
            Self::Timeout(_) => "SendTimeoutError::Timeout(..)",
            Self::Closed(_) => "SendTimeoutError::Closed(..)",
        })
    }
}

#[cfg(feature = "std")]
impl<T> fmt::Display for SendTimeoutError<T> {
```